stingray/varenergyspectrum.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import division
+import numpy as np
+from stingray.gti import check_separate, cross_two_gtis
+from stingray.lightcurve import Lightcurve
+from stingray.utils import assign_value_if_none, simon
+from stingray.crossspectrum import AveragedCrossspectrum
+import six
+
+
+class VarEnergySpectrum(object):
+    def __init__(self, events, freq_interval, energy_spec, ref_band,
+                 bin_time=1, use_pi=False, log_distr=False,
+                 segment_size=None, events2=None):
+
+        """Generic variability-energy spectrum.
+        
+        Parameters
+        ----------
+        events : stingray.events.EventList object 
+            event list
+        freq_interval : [f0, f1], floats
+            the frequency range over which calculating the variability quantity
+        energy_spec : [emin, emax, N]
```

Comment:
I have this weird sense of deja-vu, so if I've said this before and we've decided against it, let me know: I feel like it would be more useful for the user to specify the boundaries of the energy bands directly. My suggestion for this would be to make it a keyword argument, and if `None`, it will compute the variance-energy spectrum for each channel individually. If it's not `None`, it must be a list of lower/upper bin edges (for `N` energy bands there need to be `N+1` boundaries. This would get rid of at least one extra keyword argument, and give the user more flexibility in defining those energy bands. I can think of several situations were they might want to specifically use particular bands (say, if they're interested in a spectral line) that are more complicated than linear and logarithmic bands allow.